### DISCUSSION TITLE
AVD-AWS-0057 has a case that we don't want to detect in the action of clouwatch logs.

### DISCUSSION THREAD

**Discussion by int-tt** [aquasecurity/trivy#5006] [Ideas]
[AVD-AWS-0057](https://avd.aquasec.com/misconfig/aws/iam/avd-aws-0057) detects policies that write to cloudwatch logs.
https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/iam-identity-based-access-control-cwl.html#managed-policies-cwl
```
{
   "Version":"2012-10-17",
   "Statement":[
      {
      "Action": [
        "logs:CreateLogStream",
        "logs:DescribeLogStreams",
        "logs:PutLogEvents",
        "logs:GetLogEvents"
      ],
      "Effect": "Allow",
      "Resource": "arn:aws:logs:us-west-2:123456789012:log-group:SampleLogGroupName:*"
      }
   ]
}
```

Since the log-stream name is essentially a random string, it does not matter that it is specified with an asterisk.

I would like to discuss this issue!

I would register `"logs:CreateLogGroup"` and `"logs:DeleteLogGroup"` as wildcard enabled actions.
For the other actions in cloudwatch logs, I'm thinking it would be better to inspect if the action is limited to the minimum required resources in the same way as S3.

https://github.com/aquasecurity/defsec/blob/master/pkg/providers/aws/iam/wildcards.go#L20
https://github.com/aquasecurity/defsec/blob/master/rules/cloud/policies/aws/iam/no_policy_wildcards.go#L123
The reference implementation is the following commit.
https://github.com/int-tt/defsec/commit/edcc9434a6ff59c2dba56f23484a4889602b6967

Are there any other good ways to handle this?

**Comment by simar7** [comment]
hi @int-tt - thanks for the discussion. I took a look at your reference commit and it makes sense to me. We would love to review your PR if you decide to contribute to defsec for this. Thanks.

**Comment by int-tt** [comment]
Thanks for the confirmation, I created a PR!
https://github.com/aquasecurity/defsec/pull/1429

**Comment by simar7** [comment]
Closing as https://github.com/aquasecurity/trivy/issues/5010 is created to track it.
